oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -49,10 +59,46 @@ struct NodeInfo {
     handles: HashSet<Handle>,
 }
 
+impl std::fmt::Debug for NodeInfo {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "NodeInfo {{'{}', label={:?}, handles=[",
+            self.pretty_name, self.label
+        )?;
+        for (i, handle) in self.handles.iter().enumerate() {
```

Comment:
alternatively you could build the string in a more functional way and just print it out:

```rust
self.handles.iter().map(|handle| format!("{:?}", handle)).collect().join(", ");
```